src/components/molecules/Visualizer/Engine/Cesium/hooks.ts

```diff
@@ -163,6 +194,185 @@ export default ({
+      !viewer.scene ||
+      viewer.isDestroyed() ||
+      !property?.cameraLimiter ||
+      !property?.cameraLimiter.cameraLimitterTargetArea ||
+      !geodsic
+    )
+      return;
+
+    const camera = new CesiumCamera(viewer.scene);
+    camera.setView({
+      destination: Cartesian3.fromDegrees(
+        property.cameraLimiter.cameraLimitterTargetArea.lng,
+        property.cameraLimiter.cameraLimitterTargetArea.lat,
+        property.cameraLimiter.cameraLimitterTargetArea.height,
+      ),
+      orientation: {
+        heading: property?.cameraLimiter?.cameraLimitterTargetArea.heading,
+        pitch: property?.cameraLimiter?.cameraLimitterTargetArea.pitch,
+        roll: property?.cameraLimiter?.cameraLimitterTargetArea.roll,
+        up: camera.up,
+      },
+    });
+
+    const computedViewRectangle = camera.computeViewRectangle();
+    if (!computedViewRectangle) return;
+    const rectangleHalfWidth = Rectangle.computeWidth(computedViewRectangle) * Math.PI * 1000000;
+    const rectangleHalfHeight = Rectangle.computeHeight(computedViewRectangle) * Math.PI * 1000000;
+
+    const {
+      cameraLimitterTargetWidth: width = targetWidth,
+      cameraLimitterTargetLength: length = targetLength,
+    } = property?.cameraLimiter ?? {};
+
+    const { cartesianArray } = calcBoundaryBox(
+      geodsic,
+      length / 2 + rectangleHalfHeight,
+      width / 2 + rectangleHalfWidth,
+    );
+
+    setCameraViewOuterBoundaries(cartesianArray);
+  }, [property?.cameraLimiter, geodsic, camera]);
+
+  // Manage camera limiter conditions
+  useEffect(() => {
+    const camera = getCamera(cesium?.current?.cesiumElement);
+    const viewer = cesium?.current?.cesiumElement;
+    if (
+      !viewer ||
+      viewer.isDestroyed() ||
+      !property?.cameraLimiter?.cameraLimitterEnabled ||
+      !limiterDimensions
+    )
+      return;
+    if (camera) {
+      const cameraPosition = Cartographic.fromDegrees(camera?.lng, camera?.lat, camera?.height);
+
+      const destination = new Cartographic(
+        Math.clamp(
+          cameraPosition.longitude,
+          limiterDimensions.cartographicDimensions.leftDemention.longitude,
+          limiterDimensions.cartographicDimensions.rightDemention.longitude,
+        ),
+        Math.clamp(
+          cameraPosition.latitude,
+          limiterDimensions.cartographicDimensions.bottomDemention.latitude,
+          limiterDimensions.cartographicDimensions.topDemention.latitude,
+        ),
+        cameraPosition.height,
+      );
+
+      viewer.camera.setView({
+        destination: Cartographic.toCartesian(destination),
+        orientation: {
+          heading: viewer.camera.heading,
+          pitch: viewer.camera.pitch,
+          roll: viewer.camera.roll,
+          up: viewer.camera.up,
+        },
+      });
+    }
+  }, [camera, property?.cameraLimiter, limiterDimensions]);
+
+  const cameraViewBoundariesMaterial = new PolylineDashMaterialProperty({
+    color: Color.RED,
+  });
```

Comment:
Define it outside component and hooks. This still recreates cameraViewBoundariesMaterial every render time.